### ISSUE TITLE
[bug] `immutability`: false positive on DOM event target mutation

### ISSUE THREAD

**Issue by silverwind** [Rel1cx/eslint-react#1626]
*This issue was written by Claude, an AI assistant.*

## Rule

`react/immutability`

## Description

The rule flags mutations of native DOM element properties accessed via `e.currentTarget` in event handlers as immutability violations. These are standard DOM operations, not React state/props mutations.

## Repro

```tsx
import {type KeyboardEvent} from "react";

export function SearchInput() {
  function onKeyDown(e: KeyboardEvent<HTMLInputElement>) {
    if (e.key === "Escape") {
      e.currentTarget.value = ""; // false-positive: react/immutability
    }
  }

  return <input onKeyDown={onKeyDown} />;
}
```

## Expected

No error. `e.currentTarget` is a native DOM element, not a React-managed value.

## Actual

```
Do not mutate 'e.currentTarget' directly. Props and state are immutable — create a new object instead  react/immutability
```
